File: 1.0/src/cz/mg/vulkan/PFNvkBindImageMemory.java
package cz.mg.vulkan;

public class PFNvkBindImageMemory extends VkFunctionPointer {
    public PFNvkBindImageMemory() {
    }

    protected PFNvkBindImageMemory(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkBindImageMemory(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkBindImageMemory(long value) {
        setValue(value);
    }

    public PFNvkBindImageMemory(VkInstance instance) {
        super(instance, new VkString("vkBindImageMemory"));
    }

    public int call(VkDevice device, VkImage image, VkDeviceMemory memory, long memoryOffset){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), image != null ? image.getVkAddress() : VkPointer.getNullAddressNative(), memory != null ? memory.getVkAddress() : VkPointer.getNullAddressNative(), memoryOffset);
    }

    protected static native int callNative(long vkaddress, long device, long image, long memory, long memoryOffset);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkBindImageMemory.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkBindImageMemory_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong image, jlong memory, jlong memoryOffset){
    (void)env;
    (void)jc;
    PFN_vkBindImageMemory f = (PFN_vkBindImageMemory)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkImage*)jniLongToPointer(image)),
        *((VkDeviceMemory*)jniLongToPointer(memory)),
        (VkDeviceSize)memoryOffset
    );
}

